scipy/linalg/tests/test_lapack.py

```diff
@@ -488,6 +490,143 @@ def test_sing_val_update(self):
+                           [1.62+3.68j, -2.77-1.93j, 0, 0]],
+                          dtype=dtype)
+            b = np.array([[-8.86 - 3.88j, -24.09 - 5.27j],
+                          [-15.57 - 23.41j, -57.97 + 8.14j],
+                          [-7.63 + 22.78j, 19.09 - 29.51j],
+                          [-14.74 - 2.40j, 19.17 + 21.33j]],
+                         dtype=dtype)
+            x_out = np.array([[2j, 1 + 5j],
+                              [1 - 3j, -7 - 2j],
+                              [-4.001887 - 4.988417j, 3.026830 + 4.003182j],
+                              [1.996158 - 1.045105j, -6.103357 - 8.986653j]],
+                             dtype=dtype)
+        else:
+            raise ValueError(f"Datatype {dtype} not understood.")
+
+        tbtrs = get_lapack_funcs(('tbtrs'), dtype=dtype)
+        x, info = tbtrs(ab=ab, b=b, uplo='L')
+        assert_equal(info, 0)
+        assert_allclose(x, x_out, rtol=0, atol=1e-5)
+
+    @pytest.mark.parametrize('dtype,trans',
+                             [(dtype, trans)
+                              for dtype in DTYPES for trans in ['N', 'T', 'C']
+                              if not (trans == 'C' and dtype in REAL_DTYPES)])
+    @pytest.mark.parametrize('uplo', ['U', 'L'])
+    @pytest.mark.parametrize('diag', ['N', 'U'])
+    def test_random_matrices(self, dtype, trans, uplo, diag):
+        seed(1724)
+        # lda, ldb, nrhs, kd are used to specify A and b.
+        # A is of shape lda x ldb with kd super/sub-diagonals
+        # b is of shape ldb x nrhs matrix
+        lda, ldb, nrhs, kd = 4, 4, 3, 2
+        ldab = kd + 1
+
+        # Construct the diagonal and kd super/sub diagonals of A with
+        # the corresponding offsets.
+        band_widths = range(lda, lda - kd - 1, -1)
+        band_offsets = np.arange(ldab) if uplo == 'U' else np.arange(ldab) * -1
+        if dtype in REAL_DTYPES:
+            b = rand(ldb, nrhs).astype(dtype)
+            bands = [rand(width).astype(dtype) for width in band_widths]
+        elif dtype in COMPLEX_DTYPES:
+            b = (rand(ldb, nrhs) + rand(ldb, nrhs) * 1j).astype(dtype)
+            bands = [(rand(width) + rand(width) * 1j).astype(dtype)
+                     for width in band_widths]
+
+        if diag == 'U':  # A must be unit triangular
+            bands[0] = np.ones(lda, dtype=dtype)
+
+        # Construct the diagonal banded matrix A from the bands and offsets.
+        a = sps.diags(bands, band_offsets, format='dia')
+        ab = np.flipud(a.data) if uplo == 'U' else a.data
+
+        tbtrs = get_lapack_funcs('tbtrs', dtype=dtype)
+        x, info = tbtrs(ab=ab, b=b, uplo=uplo, trans=trans, diag=diag)
+        assert_equal(info, 0)
+
+        if trans == 'N':
+            assert_allclose(a @ x, b, rtol=5e-5)
+        elif trans == 'T':
+            assert_allclose(a.T @ x, b, rtol=5e-5)
+        elif trans == 'C':
+            assert_allclose(a.H @ x, b, rtol=5e-5)
```

Comment:
This line is incorrect, it should be `a.conjugate()` instead of `a.H`. When I try to use the conjugate the returned solution is incorrect (not just due to numerical errors).

Reply:
It's a conjugate _transpose_, so shouldn't_ it be `a.conjugate().T`?

Reply:
Which is equivalent to ``a.H`` and probably why ``a.H`` is *spuriously* passing. I feel a bit silly now 😅

Reply:
Thanks I feel a bit silly now 😅. ``a.conjugate().T`` is equivalent to ``a.H`` so that is why it is *spuriously* passing.

Reply:
I don't understand why it's passing at all, as NumPy arrays don't have an `H` attribute.

Reply:
It's a [``dia_matrix``](https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.dia_matrix.html). I randomly generated the "bands" (its a `tb` matrix) then use that to form the matrix and the compacted storage.

Reply:
I see. I was thinking of `ab`.

Reply:
``ab`` is used in the call to `tbtrs`. This line checks the results of the previous call using `a` and `b`